Document Service Traffic Policies (#29121)
kubernetes/website: content/en/docs/concepts/services-networking/service.md

If you try to create a Service with an invalid clusterIP address value, the API
server will return a 422 HTTP status code to indicate that there's a problem.

> ## Traffic Policies

Reviewer: style nit: this should be sentence case
```suggestion
## Traffic policies
```

> ### External Traffic Policy

Reviewer: style nit: this should be sentence case
```suggestion
### External traffic policy
```

You can set the `spec.externalTrafficPolicy` field to control how traffic from external sources is routed.
Valid values are `Cluster` and `Local`. Set the field to `Cluster` to route external traffic to all ready endpoints
and `Local` to only route to ready node-local endpoints. If the traffic policy is `Local` and there are are no node-local

> endpoints, traffic is dropped by kube-proxy.

Reviewer: ```suggestion
endpoints, the kube-proxy does not forward any traffic for the relevant Service.
```
I presume the exact behavior in this case depends on whether you're using kube-proxy in ipvs, iptables or userland mode. Is that right?

{{< note >}}

> {{< feature-state for_k8s_version="v1.22" state="alpha" >}}
> If the feature gate ProxyTerminatingEndpoints is enabled from kube-proxy, it will route traffic to terminating endpoints
> if and only if the external traffic policy is `Local` and there are only terminating endpoints on the node. This capability

Reviewer: ```suggestion
If you enable the `ProxyTerminatingEndpoints`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
`ProxyTerminatingEndpoints` for the kube-proxy, the kube-proxy checks if the node
has local endpoints and whether or not all the local endpoints are marked as terminating.

If there are local endpoints and **all** of those are terminating, then the kube-proxy ignores
any external traffic policy of `Local`. Instead, whilst the node-local endpoints remain as all
terminating, the kube-proxy forwards traffic for that Service to healthy endpoints elsewhere,
as if the external traffic policy were set to `Cluster`.

This forwarding behavior for terminating endpoints
```
```

> was added to allow external load balancers to gracefully drain connections from node ports even when the health check node

Reviewer: ```suggestion
exists to allow external load balancers to gracefully drain connections that are backed by
`NodePort` Services, even when the health check node
```

port starts to fail. Otherwise, traffic can be lost between the time a node is still in the node pool of a load balancer
and traffic is being dropped during the termination period of a pod.
{{< /note >}}

> ### Internal Traffic Policy

Reviewer: style nit: this should be sentence case
```suggestion
### Internal traffic policy
```
